src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -705,25 +705,25 @@ XamlObjectDefinition topLevelControl
 
 		private void BuildSingleTimeInitializer(IIndentedStringBuilder writer, string propertyType, string propertyName, Action propertyBodyBuilder, bool isStatic = true)
 		{
+			// The property type may be partially qualified, try resolving it through FindType
+			var propertySymbol = FindType(propertyType);
+			propertyType = propertySymbol?.GetFullName() ?? propertyType;
+
 			var sanitizedPropertyName = SanitizeResourceName(propertyName);
 
 			var propertyInitializedVariable = "_{0}Initialized".InvariantCultureFormat(sanitizedPropertyName);
 			var backingFieldVariable = "__{0}BackingField".InvariantCultureFormat(sanitizedPropertyName);
 			var staticModifier = isStatic ? "static" : "";
 			var publicPropertyType = FindType(propertyType)?.DeclaredAccessibility == Accessibility.Public
-				? GetGlobalizedTypeName(propertyType)
-				: "object";
-
-			// The property type may be partially qualified, try resolving it through FindType
-			var propertySymbol = FindType(propertyType);
-			propertyType = propertySymbol?.GetFullName() ?? propertyType;
+				? propertyType
+				: "System.Object";
 
 			writer.AppendLineInvariant($"private {staticModifier} bool {propertyInitializedVariable} = false;");
 			writer.AppendLineInvariant($"private {staticModifier} {GetGlobalizedTypeName(propertyType)} {backingFieldVariable};");
 
 			writer.AppendLine();
 
-			using (writer.BlockInvariant($"public {staticModifier} {publicPropertyType} {sanitizedPropertyName}"))
+			using (writer.BlockInvariant($"public {staticModifier} {GetGlobalizedTypeName(publicPropertyType)} {sanitizedPropertyName}"))
 			{
 				using (writer.BlockInvariant("get"))
```

Comment:
❓ Folks how would you recommend adding tests for this change?

Reply:
In this case, the best way to do that is to create a `DataTemplate` type in the same namespace as the `XamlGenerationTests` project default namespace and see if that fails.

Reply:
Yep. To elaborate on the `XamlGenerationTests` project:
1. It's built by the CI
2. It runs Uno's XAML generation
3. If it fails, the CI build fails

Reply:
Great insights. Thankyou

Reply:
🎗 Add a DataTemplate class that would conflict with the WinUI DataTemplate because of the current namespace. e.g. make an empty `DataTemplate` class in the `XamlGenerationTests` namespace (the default namespace for the current assembly, used by the generator)

Reply:
Split this out as an item for me to do in the morning. Let's get this merged as is now that Jerome can progress.